lib/mapping/CMapDefines.h

```diff
@@ -80,7 +80,7 @@ struct DLL_LINKAGE TerrainTile
 	EDiggingStatus getDiggingStatus(const bool excludeTop = true) const;
 	bool hasFavorableWinds() const;
 
-	ETerrainType terType;
+	CTerrainType terType;

```

Comment:
This line completely breaks backward compatibility for saves. If that is intentional all compatibility handling code should be cleaned up, `SERIALIZATION_VERSION` & `MINIMAL_SERIALIZATION_VERSION` constants should be changed.
(`SERIALIZATION_VERSION` should be changed even if compatibility is retained)

Reply:
I think for such a big feature it is ok to completely break backward compatibility. Any objections? We will merge it to develop only after it is completely ready. For now feature builds will be ok.

Reply:
I think it's not hard to ensure backward compatibility at this point: if version of BS < current => read old ETerrainType and create new CTerrainType instance using that old -style value, otherwise, just read  CTerrainType object

So, please think about it before merge...

Reply:
@alexvins , thank you, didn't know about it.
@nullkiller , I think for feature branch we can proceed without backward compatibility, but before merging into develop I'll try to implement something similar to what @ShubusCorporation suggested

Reply:
Investigation shown that probably we should sacrifice backward compatibility for saves

Reply:
Yes it is highly likely we should. Anyway we can merge this PR anytime you want as it does not target develop. Just ensure you track somehow all important points spotted that will be addressed later so that they are not get lost.